python/paddle/distributed/auto_parallel/operators/common.py

```diff
@@ -114,3 +114,44 @@ def find_best_compatible_distributed_operator_impl(name, op_dist_attr,
         best_compatible_impl, idx = None, -1
 
     return best_compatible_impl, idx
+
+
+def set_distributed_attr_for_var(src_op_dist_attr, var, src_var):
+    import copy
+
+    auto_paralle_context = src_op_dist_attr.get_owner_context()
+    dist_attr = copy.deepcopy(
+        auto_paralle_context.get_tensor_distributed_attr_for_program(src_var))
+    dist_attr._owner_tensor = var
+    dist_attr._owner_context = auto_paralle_context.get_tensor_distributed_attr_for_program(
+        src_var)._owner_context
+    auto_paralle_context.set_tensor_distributed_attr_for_program(var, dist_attr)
+
+
+def set_distributed_attr_for_dist_op(dist_op, dst_block, src_op_dist_attr):
```

Comment:
the same as above one, but here you just copy partial of the dist_attr info for the src

Reply:
done.